File: gfg/basic/strong-numbers4336/solution.py
class Solution:
    def isStrong(self, n):
        # code here
        s = str(n)
        ans = 0
        
        for ch in s:
            digit = int(ch)
            fact = 1
            for i in range(1,digit+1):
                fact*=i
            ans += fact
            
        return ans == n

File: gfg/basic/strong-numbers4336/solution.cpp
class Solution {
  public:
  public:
    bool isStrong(int n) {
        // code here.
        int ans = 0;
        string s = to_string(n);
        for (char ch : s){
            int digit = (ch - '0');
            int fact = 1;
            for (int i=1; i<=digit; i++){
                fact *= i;
            }
            ans += fact;
        }
        
        return ans == n;
    }
};
